ga4gh/cli.py

```diff
@@ -223,23 +261,77 @@ class ReadsSearchRunner(AbstractSearchRunner):
     """
     Runner class for the reads/search method
     """
+    class GASearchReadsRequestGoogle(protocol.ProtocolElement):
+
+        __slots__ = ['end', 'pageSize', 'pageToken', 'readGroupIds',
+                     'referenceName', 'start']
+
+        def __init__(self):
+            self.end = None
+            self.pageSize = None
+            self.pageToken = None
+            self.readGroupIds = []
+            self.referenceName = None
+            self.start = 0
+
     def __init__(self, args):
         super(ReadsSearchRunner, self).__init__(args)
-        request = protocol.GASearchReadsRequest()
+        if self.usingWorkaroundsFor(client.HTTPClient.workaroundGoogle):
+            # google says referenceId not a valid field
+            request = self.GASearchReadsRequestGoogle()
+        else:
+            request = protocol.GASearchReadsRequest()
         setCommaSeparatedAttribute(request, args, 'readGroupIds')
         request.start = args.start
         request.end = args.end
         request.referenceId = args.referenceId
         request.referenceName = args.referenceName
-        if self.usingWorkaroundsFor(client.HTTPClient.workaroundGoogle):
-            # google says referenceId not a valid field
-            del request.__dict__['referenceId']
         self.setHttpClient(request, args)
 
     def run(self):
         self._run('searchReads', 'id')
 
 
+class ReferenceBasesSearchRunner(AbstractSearchRunner):
```

Comment:
I think it would be clearer to use the patterns `SearchXRunner`, and `GetXRunner` to map to the search and get API methods.